### PR TITLE
OpenCl kernel compilation errors for android #51

### PR DESCRIPTION
**Pull request by DVEfremov**
- add ability to disable DOUBLE SUPPORT with compilation flag
  DISABLE_DOUBLE_SUPPORT
  (hardcoded DOUBLE_SUPPORT_AVAILABLE  doesn't work at least for Mali GPU at caffe time runtime)

### CODE REVIEW

**Comment by naibaf7** [comment]
You can't edit cl_kernels.cpp like this, it is not persistent at the moment. Please edit the bash script that edits these files instead.
https://github.com/naibaf7/caffe/blob/master/src/caffe/greentea/cl_kernels.sh

**Comment by DVEfremov** [comment]
Ok,  applied couple of experiments related to modification of cl_kernels.sh
I've modified
../src/caffe/greentea/cl_kernels.sh
and 
../src/caffe/greentea/cl_headers/header.cl

and build caffe as usual 
but can't see if cl_kernels.sh has been run  

cl_kernels.cpp also wasn't modified ....

Could you suggest the right way to modify and make sure my changes applied ok to cl_kernels.cpp ?

**Comment by naibaf7** [comment]
It will only be executed with the makefile scripts, not with the CMake system. It will also not work on OS X or Windows, development of that part has to happen on Linux. This is why the intermediate files are also in the github repository, even though they are supposed to be autogenerated. You have to execute it manually from the root folder before running CMake.

File: src/caffe/greentea/cl_kernels.cpp [hunk 1]
```diff
@@ -6,13 +6,18 @@
 #include <string>
 #include <type_traits>
 #include <vector>
+#ifdef DISABLE_DOUBLE_SUPPORT
```
Comment (DVEfremov) [inline src/caffe/greentea/cl_kernels.cpp:9]: here and below are generated code using make as suggested

**Comment by DVEfremov** [comment]
@naibaf7  Thanks a lot! 
Now I see ...
I've applied changes as you suggested. 
Also I've generated cl_kernels.cpp based on last changes as committed as well

**Pull request merged**